src/interface/meshblock_data.hpp

```diff
@@ -97,10 +97,10 @@ class MeshBlockData {
     SetBlockPointer(other.get());
   }
   void SetBlockPointer(const MeshBlockData<T> &other) {
-    pmy_block = other.GetBlockPointer();
+    pmy_block = other.GetBlockSharedPointer();
```

Comment:
Since pointers and shared pointers are coexisting more closely in some cases clarity might be improved by naming shared points `spmy_*` and naming pointers `pmy_*`. I don't think this is blocking but I'm just bringing it up in case you agree.